improver/utilities/cube_extraction.py

```diff
@@ -352,16 +353,130 @@ def pressure_grid(variable_on_pressure: Cube) -> np.ndarray:
+            first = len(pressure_points) - 2
+            last = -1
+            step = -1
+            last_p_slice[pressure_axis] = slice(
+                len(pressure_points) - 1, len(pressure_points)
+            )
+        else:
+            local_increment = v_increment
+            first = 1
+            last = len(pressure_points)
+            step = 1
+            last_p_slice[pressure_axis] = slice(0, 1)
+
+        for p in range(first, last, step):
+            p_slice = [slice(None) for _ in range(data.ndim)]
+            p_slice[pressure_axis] = slice(p, p + 1)
+            data[p_slice] = np.where(
+                np.logical_or(
+                    data.mask[p_slice],
+                    np.logical_or(np.isnan(data[p_slice]), np.isinf(data[p_slice])),
+                ),
+                data[last_p_slice] + local_increment,
+                data[p_slice],
+            )
+            last_p_slice = p_slice
+
+    @staticmethod
+    def _one_column_slice(cube: Cube) -> List[slice]:
+        """Create a slice object that will yield a single column along the pressure coordinate
+        from the spatial centre of the cube. Other leading dimensions will use the zeroth
+        point."""
         (pressure_axis,) = cube.coord_dims("pressure")
-        cube_shape = np.array(cube.shape)
-        max_shape = cube_shape.copy()
-        max_shape[pressure_axis] = 1
-        max_values = np.nanmax(cube.data, axis=pressure_axis).reshape(max_shape)
-        cube.data = np.where(np.isnan(cube.data), max_values, cube.data)
+        x_point = cube.shape[-1] // 2
+        y_point = cube.shape[-2] // 2
+        one_column = [slice(0, 1) for _ in range(cube.ndim)]
+        one_column[pressure_axis] = slice(None)
+        one_column[-2] = slice(y_point, y_point + 1)
+        one_column[-1] = slice(x_point, x_point + 1)
+        return one_column
+
+    def fill_in_bounds(
+        self, result_array: np.ma.MaskedArray, source_cube: Cube
+    ) -> np.ndarray:
+        """Populate any invalid values with the maximum or minimum pressure for that column.
+        Maximum pressure is chosen if the maximum data value in the column is lower than
+        the value of self.value_of_pressure_level"""
+        pressure_coord = source_cube.coord("pressure")
+        increasing_p_order = np.all(np.diff(pressure_coord.points) > 0)
+        max_pressure = pressure_coord.points.max()
+        min_pressure = pressure_coord.points.min()
+        (pressure_axis,) = source_cube.coord_dims("pressure")
+        max_pressure_index = np.argmax(pressure_coord.points)
+        # Define a slice of one pressure column
+        one_p_column = self._one_column_slice(source_cube)
+        # Define increasing_v_order if at least half of the differences in the
+        # zeroth pressure column are positive
+        increasing_v_order = (
+            np.diff(source_cube.data[one_p_column].squeeze()) > 0
+        ).sum() > len(pressure_coord.points) / 2
```

Comment:
With a bit of hindsight, I think `increasing_v_order` ought to be a compulsory argument to the class `process` method as the calling programme will have the knowledge of what the variable cube contains and how it is expected to behave. I think a boolean describing whether the variable usually changes proportional to pressure or inversely proportional to pressure is what we need.